### PR TITLE
Copy all strategies into an environment

### PR DESCRIPTION
**Pull request by Tymek**
Add a button to an environment with no strategies, to clone all strategies from another environment.
Zrzut ekranu 2022-08-02 091411

### CODE REVIEW

**Review by NicolaeUnleash — COMMENTED** [review]
Wow this is very nice functionality

File: src/component/feature/FeatureStrategy/FeatureStrategyEmpty/CopyButton/CopyButton.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+export const CopyButton: VFC<ICopyButtonProps> = ({
+    environmentId,
+    environments,
+    onClick,
+}) => {
+    const projectId = useRequiredPathParam('projectId');
+    const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+    const open = Boolean(anchorEl);
+    const handleClick = (event: MouseEvent<HTMLButtonElement>) => {
```
Comment (FredrikOseberg) [inline src/component/feature/FeatureStrategy/FeatureStrategyEmpty/CopyButton/CopyButton.tsx:31]: We are trying to use `onHandlerName` as convention

File: src/component/feature/FeatureStrategy/FeatureStrategyEmpty/CopyButton/CopyButton.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        setAnchorEl(event.currentTarget);
+    };
+    const handleClose = () => {
```
Comment (FredrikOseberg) [inline src/component/feature/FeatureStrategy/FeatureStrategyEmpty/CopyButton/CopyButton.tsx:34]: here too

File: src/component/feature/FeatureStrategy/FeatureStrategyEmpty/CopyButton/CopyButton.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+    const enabled = environments.some(environment =>
+        hasAccess(CREATE_FEATURE_STRATEGY, projectId, environment)
+    );
+
+    return (
+        <div>
+            <Tooltip title={enabled ? '' : '(Access denied)'}>
+                <div>
+                    <Button
```
Comment (FredrikOseberg) [inline src/component/feature/FeatureStrategy/FeatureStrategyEmpty/CopyButton/CopyButton.tsx:47]: Could we use PermissionButton here?

**Review by FredrikOseberg — CHANGES_REQUESTED** [review]
Awesome functionality. Minor comments.

File: src/component/feature/FeatureStrategy/FeatureStrategyEmpty/CopyButton/CopyButton.tsx (hunk 3 above)
Comment (Tymek) [reply src/component/feature/FeatureStrategy/FeatureStrategyEmpty/CopyButton/CopyButton.tsx:47]: `PermissionButton` has built-in logic for access permission related to single item, not to an array of items.

Comment (FredrikOseberg) [reply src/component/feature/FeatureStrategy/FeatureStrategyEmpty/CopyButton/CopyButton.tsx:40]: Yeah, you're right about that.

**Review by FredrikOseberg — APPROVED** [review]
LGTM

**Pull request merged**